### DISCUSSION TITLE
Aspnet core: How do you serialize a collection of Practitioners?

### DISCUSSION THREAD

**Discussion by jamescanady** [FirelyTeam/firely-net-sdk#2283] [Q&A]
Hello,
I have ran into an issue where the MVC JsonOptions is being ignored when serializing during a MVC action response.
We're trying to return a Practitioner (other Fhir Objects) back as a response from MVC api requests.

```csharp
builder.Services.AddControllers()
  .AddJsonOptions(options =>
  {
      options.JsonSerializerOptions.ForFhir(typeof(Hl7.Fhir.Model.Practitioner).Assembly).Compact();
  })
```

 ```csharp
public IActionResult Get()
    {
        //var provider = new Data.Models.sProvider.Provider();
        //var practitioner = new Data.Models.Fhir.SymplrPractitioner(provider);
        var practitioner = new Hl7.Fhir.Model.Practitioner();
        practitioner.Name.Add(new () {
            Given = new [] { "Sally" },
            Family = "Smith",
            Period = new () {
                Start = new DateTime(2002, 01, 01).ToString()
            }
        });
        practitioner.Identifier.Add(new () {
            Use = Identifier.IdentifierUse.Official,
            Period = new () {
                End = new DateTime(2032, 01, 01).ToString()
            }
        });
        var practitioners = new [] { practitioner };
        return Ok(practitioners);
    }
```
The output we're getting isn't serialized properly, its as if the jsonOptions are being ignored for this request.

```json
[
	{
		"identifier": [
			{
				"use": {
					"value": "official"
				},
				"period": {
					"end": {
						"value": "1/1/2032 12:00:00 AM"
					}
				}
			}
		],
		"name": [
			{
				"family": {
					"value": "Smith"
				},
				"given": [
					{
						"value": "Sally"
					}
				],
				"period": {
					"start": {
						"value": "1/1/2002 12:00:00 AM"
					}
				}
			}
		]
	}
]
```

are there any suggestions on how to wire up the serialization options to allow for the correct response from the action?

**Comment by brianpos** [comment]
I haven't tried this new style format, but you could take a look at the facade project I wrote that helps with writing servers using this library
https://github.com/brianpos/fhir-net-web-api which is also on nuget
Some usage notes are here
https://brianpos.com/2022/04/19/create-your-own-net-fhir-server-fast/

**Comment by ghost** [comment]
@jamescanady  - In your code, why do you pass `typeof(Hl7.Fhir.Model.Practitioner).Assembly` ? Instead, just `.ForFhir()`.
